torch/csrc/tensor/python_tensor.cpp

```diff
@@ -176,8 +194,10 @@ static THPObjectPtr get_storage_obj(const Type& type) {
 }
 
 static void set_type(PyTensorType& type_obj, Backend backend, ScalarType scalarType) {
-  auto baseType = globalContext().type_registry[static_cast<int>(backend)][static_cast<int>(scalarType)].get();
-  type_obj.aten_type = baseType ? torch::autograd::VariableType::getType(*baseType) : nullptr;
+  // This field is lazily initialized from backend and scalar_type
```

Comment:
So, strictly speaking, the problem is if you call set_type before you initialize CUDA (and not just whether or not you loaded ATen_cuda.so). And in this case, calling set_type before initialization basically always happens, because at program initialization time, CUDA is not yet initialized.